sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -209,26 +216,148 @@ private HttpResponse receiveResponse(HttpRequest httpRequest, HttpURLConnection
+                    try {
+                        BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+                        processBuffer(reader);
+                    } catch (IOException e) {
+                        retryExceptionForSSE(e);
+                    }
+                }
+                return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders);
+            } else {
+                AccessibleByteArrayOutputStream outputStream = new AccessibleByteArrayOutputStream();
+
+                try (InputStream errorStream = connection.getErrorStream();
+                     InputStream inputStream = (errorStream == null) ? connection.getInputStream() : errorStream) {
+                    byte[] buffer = new byte[8192];
+                    int length;
+                    while ((length = inputStream.read(buffer)) != -1) {
+                        outputStream.write(buffer, 0, length);
+                    }
                 }
-            }
 
-            return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders,
-                BinaryData.fromByteBuffer(outputStream.toByteBuffer()));
+                return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders,
+                    BinaryData.fromByteBuffer(outputStream.toByteBuffer()));
+            }
         } catch (IOException e) {
             throw LOGGER.logThrowableAsError(new RuntimeException(e));
+
         } finally {
             connection.disconnect();
         }
     }
 
+    /**
+     * Processes the sse buffer and dispatches the event
+     * @param reader The BufferedReader object
+     * @throws IOException If an I/O error occurs
+     */
+    private void processBuffer(BufferedReader reader) throws IOException {
+        StringBuilder sb = new StringBuilder(), data = new StringBuilder();
+        String eventStr = null;
+        long id = -1;
+        ServerSentEvent event = new ServerSentEvent();
+        int dataRead = reader.read();
+        while (dataRead != -1) {
+            sb.append((char) dataRead);
+            dataRead = reader.read();
+            int index;
+            while ((index = sb.indexOf("\n\n")) >= 0) { // blocks of data are separated by double newlines
+                String[] lines = sb.substring(0, index).split("\n");
+                sb.delete(0, index + 2); // clear first block including "\n\n"
+                boolean hasDataOrEvent = false;
+                for (String line : lines) { //start parsing line by line
+                    int idx = line.indexOf(':');
+                    if (idx < 0) {
+                        continue; // ignore invalid data
+                    } else if (idx == 0) { // capture comment line
+                        event = event.setComment(line.substring(1).trim());
+                    }
+                    String field = line.substring(0, idx), value = line.substring(idx + 1).trim();
```

Comment:
Split this into 2 separate expressions for readability.

```suggestion
                    String field = line.substring(0, idx);
                    String value = line.substring(idx + 1).trim();
```